ddtrace/internal/flare.py

```diff
@@ -0,0 +1,184 @@
+import io
+import json
+import logging
+import os
+import secrets
+import shutil
+import tarfile
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Tuple
+
+from ddtrace import config
+from ddtrace._logger import _add_file_handler
+from ddtrace.internal.logger import _get_handler
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.utils.http import get_connection
+
+
+TRACER_FLARE_DIRECTORY = "tracer_flare"
+TRACER_FLARE_TAR = f"{TRACER_FLARE_DIRECTORY}.tar"
+TRACER_FLARE_ENDPOINT = "/tracer_flare/v1"
+TRACER_FLARE_FILE_HANDLER_NAME = "tracer_flare_file_handler"
+TRACER_FLARE_LOCK = "tracer_flare.lock"
+DEFAULT_TIMEOUT_SECONDS = 5
+
+log = get_logger(__name__)
+
+
+class Flare:
+    def __init__(self, timeout_sec: int = DEFAULT_TIMEOUT_SECONDS):
+        self.original_log_level = 0
+        self.timeout = timeout_sec
+
+    def _prepare(self, configs: List[dict]):
+        """
+        Update configurations to start sending tracer logs to a file
+        to be sent in a flare later.
+        """
+        os.makedirs(TRACER_FLARE_DIRECTORY, exist_ok=True)
+        for c in configs:
```

Comment:
This variable is used in enough lines that naming it with a single letter might become a little cumbersome.